src/circuit/gadget/ecc/chip/mul.rs

```diff
@@ -128,8 +141,6 @@ impl Config {
             .map(move |(name, poly)| (name, q_mul_lsb.clone() * poly))
         });
 
-        self.hi_config.create_gate(meta);
-        self.lo_config.create_gate(meta);
```

Comment:
Hmm, I didn't spot that there was a gate defined just above these; all three of these sub-config refactors still move across the above gate, causing a constraint reorder in every commit.

I'm going to try inserting a commit that temporarily splits up `mul::Config` into two pieces, so that the first can be refactored before this commit, to prevent re-ordering. Then the existing "`mul::Config` refactor" commit can re-assemble the pieces.

Reply:
Done in 0ede6b230144ec994a4fcbcc24ef5858f7a78084.